XenModel/Actions/HealthCheck/XenServerHealthCheckUpload.cs

```diff
@@ -66,10 +66,12 @@ public XenServerHealthCheckUpload(string token, int verbosity, string uploadUrl,
         }
 
         // Request an upload and fetch the uploading id from CIS.
-        public string InitiateUpload(string fileName, long size)
+        public string InitiateUpload(string fileName, long size, string caseNumber)
         {
             // Request a new bundle upload to CIS server.
             string FULL_URL = UPLOAD_URL + "bundle/?size=" + size.ToString() + "&name=" + fileName;

```

Comment:
Not really becasue of this change set, but in general I think special characters should be URLencoded. Like & should be %26 (this character is allowed in file names)